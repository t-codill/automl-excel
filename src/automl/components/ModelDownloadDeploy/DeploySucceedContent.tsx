
import { Link, MessageBar, MessageBarType } from "office-ui-fabric-react";
import * as React from "react";
import { BaseComponent } from "../Base/BaseComponent";

export interface IDeploySucceedContentProps {
    deployName: string;
    startTime: string;
    scoringUri: string | undefined;
}

export class DeploySucceedContent extends BaseComponent<
    IDeploySucceedContentProps,
    {}, {}>  {
    protected serviceConstructors = {};

    protected readonly getData = undefined;

    public readonly render = (): React.ReactNode => {
        return (
            <>
                <MessageBar messageBarType={MessageBarType.success}>
                    <strong>{this.props.deployName}</strong> deployed successfully. <br /><br />
                    <strong>Score URI: </strong> <Link href={this.props.scoringUri} target="_blank">{this.props.scoringUri}</Link> <br />
                    {
                        // TODO: Add this back once MMS fixed swagger.json
                        // <strong>Schema: </strong> <Link href={this.getSchemaUrl()} target="_blank">swagger.json</Link> <br /><br />
                    }
                    Deployed since {this.props.startTime}.
                </MessageBar>
            </>
        );
    }

    // TODO: Add this back once MMS fixed swagger.json
    // private readonly getSchemaUrl = (): string | undefined => {
    //     if (!this.state.scoringURI) {
    //         return undefined;
    //     }
    //     const scoringURI = this.state.scoringURI;
    //     return `${scoringURI.substring(0, scoringURI.lastIndexOf("score"))}swagger.json`;
    // }
}
