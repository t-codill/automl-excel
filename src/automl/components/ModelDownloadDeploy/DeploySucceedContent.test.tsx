import { shallow, ShallowWrapper } from "enzyme";
import * as React from "react";
import { DeploySucceedContent, IDeploySucceedContentProps } from "./DeploySucceedContent";

describe("Deploy Succeed Content", () => {
    let tree: ShallowWrapper<IDeploySucceedContentProps>;

    it("should render when scoring url is undefined", async () => {
        tree = shallow(<DeploySucceedContent
            deployName="AAA"
            startTime="test-start-time"
            scoringUri={undefined}
        />);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render with valid property", async () => {
        tree = shallow(<DeploySucceedContent
            deployName="AAA"
            startTime="test-start-time"
            scoringUri="testScoringUrl"
        />);
        expect(tree)
            .toMatchSnapshot();
    });
});
