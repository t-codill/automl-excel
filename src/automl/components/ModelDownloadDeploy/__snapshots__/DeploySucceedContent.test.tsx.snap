// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deploy Succeed Content should render when scoring url is undefined 1`] = `
<Fragment>
  <StyledMessageBarBase
    messageBarType={4}
  >
    <strong>
      AAA
    </strong>
     deployed successfully. 
    <br />
    <br />
    <strong>
      Score URI: 
    </strong>
     
    <StyledLinkBase
      target="_blank"
    />
     
    <br />
    Deployed since 
    test-start-time
    .
  </StyledMessageBarBase>
</Fragment>
`;

exports[`Deploy Succeed Content should render with valid property 1`] = `
<Fragment>
  <StyledMessageBarBase
    messageBarType={4}
  >
    <strong>
      AAA
    </strong>
     deployed successfully. 
    <br />
    <br />
    <strong>
      Score URI: 
    </strong>
     
    <StyledLinkBase
      href="testScoringUrl"
      target="_blank"
    >
      testScoringUrl
    </StyledLinkBase>
     
    <br />
    Deployed since 
    test-start-time
    .
  </StyledMessageBarBase>
</Fragment>
`;
